netctl/netctl.go

```diff
@@ -25,9 +30,99 @@ func getClient(ctx *cli.Context) *contivClient.ContivClient {
 		errExit(ctx, 1, "Error connecting to netmaster", false)
 	}
 
+	// if the netctl config exists, apply it to the client
+	if configExists(ctx) {
+		if err := applyConfig(cl); err != nil {
+			errExit(ctx, exitInvalid, "error processing netctl config: "+err.Error(), false)
+		}
+	}
+
+	// if --insecure, set a custom http.Client with cert checking disabled
+	if ctx.GlobalBool("insecure") {
+		tr := &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		}
+		httpClient := &http.Client{Transport: tr}
+
+		if err := cl.SetHTTPClient(httpClient); err != nil {
+			errExit(ctx, exitInvalid, "error setting custom HTTP client: "+err.Error(), false)
+		}
+	}
+
 	return cl
 }
 
+func readLoginCredentials(ctx *cli.Context) (string, string) {
+	fmt.Print("Username: ")
+
+	reader := bufio.NewReader(os.Stdin)
+	username, err := reader.ReadString('\n')
+	if err != nil {
+		errExit(ctx, exitIO, err.Error(), false)
+	}
+
+	username = strings.TrimSpace(username) // ReadString includes the newline
+
+	if len(username) == 0 {
+		errExit(ctx, exitInvalid, "you must specify a username", false)
+	}
+
+	fmt.Print("Password: ")
+
+	// use terminal.ReadPassword() so there's no echoing of what's typed
+	bytePassword, err := terminal.ReadPassword(0) // fd 0 = stdin
+	if err != nil {
+		errExit(ctx, exitIO, err.Error(), false)
+	}
+	password := string(bytePassword)
+	fmt.Println("")
+
+	return username, password
+}
+
+func login(ctx *cli.Context) {
```

Comment:
I was just following the convention in the rest of the file of not having comments 😉 

Done